### ISSUE TITLE
Incorrect conversion of numbered list

### ISSUE THREAD

**Issue by djm158** [BlueHatbRit/mdpdf#32]
Tried converting the following file:
https://raw.githubusercontent.com/laboon/CS1530_Fall2017/master/reading-quizzes/quiz4.md

using the command line tool. The file has a numbered list
```md
1. item one
2. item two
3. item three
```

but the converted pdf does not convert the list correctly. Instead it looks like:
```
1. item one
1. item two
1. item three
```
Attached an example. I'd be willing to look at it when I have a chance but wanted to document this behavior first. This behavior appeared in the debug html as well.

[quiz4.pdf](https://github.com/BlueHatbRit/mdpdf/files/1353025/quiz4.pdf)

**Comment by BlueHatbRit** [comment]
Thanks for reporting this one, it looks like it's probably a bug in [showdownjs](https://github.com/showdownjs/showdown) which we use for the markdown->html conversion.

It looks like there may have been a fix for this in a recent release [1.7.2](https://github.com/showdownjs/showdown/releases/tag/1.7.2). We should try and bump that dependency and see if that fixes the issue. It'd be great if you'd like to give that a go as we're currently on 1.6.x. If you don't get a chance then I'll give it a look.

**Comment by djm158** [comment]
I'll give it a go!

**Comment by djm158** [comment]
It looks like this is an issue with showdown parsing the backticks (```) in my markdown file. Unfortunately the most up to date showdown release does not have a fix for this, I'll see about creating an issue over there.

**Comment by BlueHatbRit** [comment]
That's a shame, I'll give this a test myself today and see if I can peg it down for your issue. Are you able to link this issue in your report to showdownjs?

Do you have a markdown example that illustrates it's back ticks causing the problem and not paragraph based lists (with the latest showdown)?

**Comment by djm158** [comment]
Didn't make a report yet, wanted to go through the showdown docs to see if this was valid first. You can test playing with the markdown [here](http://showdownjs.github.io/demo/). 

Also tested using showdown's cli and this test file
`Name: `Daniel McGrath` Class start time: `1:00`

1. Does every software system have an architecture?  Why or why not?

2. You are designing a climate control system, which accepts inputs from a variety of distributed sensors (e.g., thermometers, smoke sensors, barometers, etc.), and normalizes environmental conditions based on this input.  For example, if average temperature is below a threshold, the heater will be turned on.  Occasional input needs to be responded to immediately; for example, a smoke sensor detecting particulate matter may set off a fire alarm immediately.
Of the architectural styles listed in Tsui _et al_, which one would be best for this kind of system?  Why?

3. You are designing a system to store classes so that students can view information on the classes available for the next semester.  There is little to no business logic associated with this system.  The focus is just on storing and viewing (and for administrators, occasionally adding) data to the system.
Of the architectural styles listed in Tsui _et al_, which one would be best for this kind of system?  Why?

4. After reviewing your system design, your co-worker notes that you have two classes with content coupling.  Why is this problematic?

5. You have written a class whose methods are all related in terms of some control sequence.  What level of cohesion is this?
`

Using command `showdown maketml -i test.md -o test.html` renders the following output 

`<p>Name: <code>Daniel McGrath</code> Class start time: <code>1:00</code></p>
<ol>
<li><p>Does every software system have an architecture?  Why or why not?</p></li>
<li><p>You are designing a climate control system, which accepts inputs from a variety of distributed sensors (e.g., thermometers, smoke sensors, barometers, etc.), and normalizes environmental conditions based on this input.  For example, if average temperature is below a threshold, the heater will be turned on.  Occasional input needs to be responded to immediately; for example, a smoke sensor detecting particulate matter may set off a fire alarm immediately.
Of the architectural styles listed in Tsui <em>et al</em>, which one would be best for this kind of system?  Why?</p></li>
<li><p>You are designing a system to store classes so that students can view information on the classes available for the next semester.  There is little to no business logic associated with this system.  The focus is just on storing and viewing (and for administrators, occasionally adding) data to the system.
Of the architectural styles listed in Tsui <em>et al</em>, which one would be best for this kind of system?  Why?</p></li>
<li><p>After reviewing your system design, your co-worker notes that you have two classes with content coupling.  Why is this problematic?</p></li>
<li><p>You have written a class whose methods are all related in terms of some control sequence.  What level of cohesion is this?</p></li>
</ol>`

**Comment by BlueHatbRit** [comment]
@djm158 I've wrapped your snippet above in some code marks because github actually renders HTML and markdown in comments so it's hard to see the incorrect / correct results.

It looks like showdown is spitting out the correct HTML in that example, though when pushing through your original markdown example in the top issue comment I can see it's producing the bug. I think it's safe to say this is an issue in Showdown as it should be rendering the numbers correctly.

You should be able to work around this in mdpdf for now using some HTML within your markdown though this is known to a bit buggy in Showdown and I know of other projects that have issues supporting this.

**Comment by BlueHatbRit** [comment]
Just as an example of what I mean, this is what your original MD document would look like if it used some HTML to make the correction: https://gist.githubusercontent.com/BlueHatbRit/482d0ae892733e8d4a51e172264f507e/raw/d43464ebb6b60659dcb59c3f12cafad7490d42c7/example-md.md

**Comment by tivie** [comment]
# TL;DR;

It's not a bug in showdown, but rather a problem the the input. List's Sub-blocks must be indented 4 spaces (or they will be treated as a "root block")  

# Explanation 
A bunch of things are happening here:

## List's Sub-blocks must be indented 4 spaces 

According to [the original markdown spec](https://daringfireball.net/projects/markdown/syntax#list) a sub-block, in a list, must be indented to be considered part of the list.

**Example:** ([see showdown's live demo][demo1])

input MD:

    1. Does every software system have an architecture?  Why or why not?
    
        ```
        _________________________________________________________________________
        _________________________________________________________________________
        ```
    
    2. You are designing a climate control system, which accepts inputs from a variety...
    
        ```
        _________________________________________________________________________
        _________________________________________________________________________
        ```

generated HTML:

```html
<ol>
<li><p>Does every software system have an architecture?  Why or why not?</p>
<pre><code>_________________________________________________________________________
_________________________________________________________________________</code></pre></li>
<li><p>You are designing a climate control system, which accepts inputs from a variety...</p>
<pre><code>_________________________________________________________________________
_________________________________________________________________________</code></pre></li>
</ol>
```

In your md source, your code blocks are not indented so each list item is interpreted as  *"new"* list. [see demo][demo2]

This happens in every markdown implementation, even in github's implementation. ([see babelmark2 for comparison in implementations][bb2_1])

```html
<ol>
<li>Does every software system...</li>
</ol>
<pre><code>_____________________________________________________________</code></pre>
<ol>
<li>You are designing a climate control system...</li>
</ol>
```

## Your input "seems" to work in github's

Github has an extra *"feature"* where the number you in the first list item counts as the starting value of the ordered list. For instance, copy/paste the following markdown in a comment here in github and click preview:

Input:

```md
2. foo
6. bar
8. baz
9. bazinga
```

Output:

2. foo
6. bar
8. baz
9. bazinga

This feature is non standard and not implemented in showdown (although it's planned for a soonish future release),

So, in your MD, every list item is considered a new list, rather than a list item. But since github supports defining the starting number of ordered lists, the output seems correct (although it's not)


[demo1]: http://showdownjs.github.io/demo/#/1.%20Does%20every%20software%20system%20have%20an%20architecture%3F%20%20Why%20or%20why%20not%3F%0A%0A%20%20%20%20%60%60%60%0A%20%20%20%20_________________________________________________________________________%0A%20%20%20%20_________________________________________________________________________%0A%20%20%20%20%60%60%60%0A%0A2.%20You%20are%20designing%20a%20climate%20control%20system%2C%20which%20accepts%20inputs%20from%20a%20variety...%0A%0A%20%20%20%20%60%60%60%0A%20%20%20%20_________________________________________________________________________%0A%20%20%20%20_________________________________________________________________________%0A%20%20%20%20%60%60%60

[demo2]: http://showdownjs.github.io/demo/#/1.%20Does%20every%20software%20system%20have%20an%20architecture%3F%20%20Why%20or%20why%20not%3F%0A%0A%60%60%60%0A_________________________________________________________________________%0A_________________________________________________________________________%0A%60%60%60%0A%0A2.%20You%20are%20designing%20a%20climate%20control%20system%2C%20which%20accepts%20inputs%20from%20a%20variety...%0A%0A%60%60%60%0A_________________________________________________________________________%0A_________________________________________________________________________%0A%60%60%60

[bb2_1]: https://johnmacfarlane.net/babelmark2/?normalize=1&text=1.+Does+every+software+system+have+an+architecture%3F++Why+or+why+not%3F%0A%0A%60%60%60%0A________________________________________________________________________%0A%0A%60%60%60%0A%0A2.+You+are+designing+a+climate+control+system%2C+which+accepts+inputs+from+a+variety+of+distributed+sensors+(e.g.%2C%0AOf+the+architectural+styles+listed+in+Tsui+_et+al_%2C+which+one+would+be+best+for+this+kind+of+system%3F++Why%3F%0A%0A%60%60%60%0A________________________________________________________________________%0A%0A________________________________________________________________________%0A%60%60%60

**Comment by BlueHatbRit** [comment]
Hi @tivie, thanks so much for stopping by and explaining this in such great detail. I apologise for not reading the spec initially to check this. Seems we were also fooled by github's implementation of the planned / non-standard feature.

In light of that information, the solution in the TLDR seems to work just fine with mdpdf so I'm going to close this issue.

Also thanks for raising it @djm158, I learned something new from it.
